cpp/src/arrow/util/value_parsing.h

```diff
@@ -55,26 +55,18 @@ class ARROW_EXPORT TimestampParser {
 
 namespace internal {
 
-/// \brief A class providing conversion from strings to some Arrow data types
-///
-/// Conversion is triggered by calling operator().  It returns true on
-/// success, false on failure.
-///
-/// The class may have a non-trivial construction cost in some cases,
-/// so it's recommended to use a single instance many times, if doing bulk
-/// conversion. Instances of this class are not guaranteed to be thread-safe.
-///
-template <typename ARROW_TYPE, typename Enable = void>
-class StringConverter;
+namespace detail {
 
-template <>
-class StringConverter<BooleanType> {
- public:
-  explicit StringConverter(const std::shared_ptr<DataType>& = NULLPTR) {}
+template <typename ARROW_TYPE>
+struct StringConverter {
+  static bool Convert(const char*, size_t, void*) { return false; }
```

Comment:
This allows construction of an always-fails StringConverter for any arrow type, replacing the previous behavior under which `StringConverter<T>` was a compile error if conversion for `T` is not defined. This replacement seems extremely dubious to me, especially if not clearly documented. Please revert this and any instantiations of `StringConverter<T>` or `ParseValue<T>` which are undefined.